### PR TITLE
Enable users to set the gRPC max message size

### PR DESCRIPTION
**Pull request by taeyeon-Kim**
resolved: #64

### CODE REVIEW

**Review by sohyun-ku — APPROVED** [review]

**Review by kojandy — APPROVED** [review]

File: doc/installation.md [hunk 1]
```diff
@@ -36,6 +36,10 @@ You can override any configuration values with `-D` option.
 To change the http(default 8080) and grpc(default 8080) ports used by Collector, you can change the settings below.
  - `-Darmeria.port`: http, grpc uses 8080 by default
 
+To change the default value of 10MB for the max message size of gRPC in Collector, change the setting below.
```
Comment (junoyoon) [inline doc/installation.md:39]:
```suggestion
To change the the max message size of gRPC in Collector, change the setting below. 
```

File: doc/installation.md [hunk 2]
```diff
@@ -174,6 +178,10 @@ packages=com.navercorp
 
 - Setting up the agent via `-javaagent:path/to/scavenger-agent.jar`
 - Specify the config file via `-Dscavenger.configuration=path/to/scavenger.conf`
+- To change the default value of 10MB for the max message size of gRPC in agent, change the setting below.
```
Comment (junoyoon) [inline doc/installation.md:181]:
```suggestion
- To change the max message size of gRPC in agent, change the setting below.
```

File: doc/installation.md [hunk 3]
```diff
@@ -174,6 +178,10 @@ packages=com.navercorp
...
+- `-Dscavenger.max-message-size`
+    - Supports `B`, `KB`, `MB`, `GB`, and `TB` units.
+> **We recommend tuning your codeBase configuration before changing `scavenger.max-message-size`.**
```
Comment (junoyoon) [inline doc/installation.md:184]:
```suggestion
> **We recommend tuning your codeBase configuration for minimizing tracing ranges before changing `scavenger.max-message-size`.**
```

File: scavenger-agent-java/src/main/java/com/navercorp/scavenger/javaagent/publishing/GrpcClient.java [hunk 4]
```diff
@@ -66,10 +75,11 @@ private void createNewChannelIfShutdown() {
         if (channel == null || channel.isShutdown()) {
             log.fine("[scavenger] creating new grpc channel");
             try {
-                channel = createChannel();
+                int maxMessageSize = System.getProperty(MAX_MESSAGE_SIZE_CONFIG) != null ? maxMessageSize() : DEFAULT_MAX_MESSAGE_SIZE;
```
Comment (junoyoon) [inline scavenger-agent-java/src/main/java/com/navercorp/scavenger/javaagent/publishing/GrpcClient.java:78]: Isn't it better to implement ```System.getProperty(MAX_MESSAGE_SIZE_CONFIG) != null ? maxMessageSize() : DEFAULT_MAX_MESSAGE_SIZE;``` in maxMessageSize()?

File: doc/installation.md (hunk 1 above)
Comment (taeyeon-Kim) [reply doc/installation.md]: fixed it in 755bb3d91980ac24c9572a060398d1c60a1b5bd7

File: doc/installation.md (hunk 2 above)
Comment (taeyeon-Kim) [reply doc/installation.md]: fixed it in 755bb3d91980ac24c9572a060398d1c60a1b5bd7

File: doc/installation.md (hunk 3 above)
Comment (taeyeon-Kim) [reply doc/installation.md]: fixed it in 755bb3d91980ac24c9572a060398d1c60a1b5bd7

File: scavenger-agent-java/src/main/java/com/navercorp/scavenger/javaagent/publishing/GrpcClient.java (hunk 4 above)
Comment (taeyeon-Kim) [reply scavenger-agent-java/src/main/java/com/navercorp/scavenger/javaagent/publishing/GrpcClient.java:78]: fixed it in 755bb3d91980ac24c9572a060398d1c60a1b5bd7

File: doc/installation.md [hunk 5]
```diff
@@ -36,6 +36,10 @@ You can override any configuration values with `-D` option.
 To change the http(default 8080) and grpc(default 8080) ports used by Collector, you can change the settings below.
  - `-Darmeria.port`: http, grpc uses 8080 by default
 
+To change the the max message size of gRPC in Collector, change the setting below.
```
Comment (junoyoon) [inline doc/installation.md:39]:
```suggestion
To change the max message size of gRPC in Collector, change the setting below.
```

File: scavenger-agent-java/src/main/java/com/navercorp/scavenger/javaagent/publishing/GrpcClient.java (hunk 4 above)
Comment (junoyoon) [reply scavenger-agent-java/src/main/java/com/navercorp/scavenger/javaagent/publishing/GrpcClient.java:78]: int maxMessageSize =  maxMessageSize()

Comment (taeyeon-Kim) [reply scavenger-agent-java/src/main/java/com/navercorp/scavenger/javaagent/publishing/GrpcClient.java]: oh... i missed it... ammend in 6722455b1b873b57577794d5338d447041462e86

**Review by junoyoon — APPROVED** [review]

**Pull request merged**